### ISSUE TITLE
Question: How would you add your own shortcuts? 

### ISSUE THREAD

**Issue by geoxcaliber** [glenviewjeff/HyperWindowSnap#9]
I like all the current snapping features, but I would like to add in more snapping, like into thirds and other things. Do you know if it's possible? 
I got this off a forum, but when it snaps it doesn't snap to a predefined place, like left, right, etc. If it's on the right, it does the action on the right. 

#!u::ResizeWin(0,600)
ResizeWin(Width = 0,Height = 0)
{
  WinGetPos,X,Y,W,H,A
  If %Width% = 0
    Width := W
  If %Height% = 0
    Height := H
  WinMove,A,,%X%,%Y%,%Width%,%Height%
}

I was hoping I would be able to use a simple command like Resize (Height, Width) and then customize the height width on my own. If it's not possible, could you explain how you did it in your script? I saw some commands that specified Left and Right, so is there a way I can like map the hotkeys to the left/right, and it'll go there.

**Comment by glenviewjeff** [comment]
Yes, there's an issue I opened to add a feature to switch into a mode where it would divide by thirds. But read the instructions. You can further subdivide windows with the script as-is, and "scoot" them around the screen.

**Comment by geoxcaliber** [comment]
But the mode is not available yet right? Also, scooting and shrinking is not very good right now/useful. Let's say you snap a window to the left (Win + Numpad4) if you Alt + Win + Numpad4 it, it'll shrink to about half and it'll go to the very left. If you do the same but this time do Numpad6, it'll still be in the left side but not to the very left. You can then do Numpad6 (Snapping) and do everything again but this time in the right, this gives you four windows. However, it's not often I need four windows and when I do, I snap each one to a corner. Dividing windows into thirds would give each more space. Here's what can be done to make better use of shrinking. Since at the moment Win + Numpad5 is not in use, it can put a window into a "priority". It's a template that fancyzones has -> has https://i.imgur.com/OXPaGg7.png. Using powertoys to put browser into the middle (biggest one) I've been able to accomplish basically the priority grid with shrinking windows -> https://i.imgur.com/HCRi6aQ.png. The two on the outside are a snap numpad4 and 6, and then alt win 4 and alt win 6. Also, what you had in mind is pretty nice too. Win + NumpadDiv toggles a whole new 9 key setup. Press again to disable. Shift + Win would also be good. After Div or Shift, 123 would be the thirds. 1 is the left, 2, middle, 3 right. 7 8 9 can also be rows. Still thirds, but this time in rows. As for 4 5 6, idk. What you can do is make 5 maximize, pressing it again will bring it to the middle as a small square window. Kind of like this -> https://i.imgur.com/S3ldxHE.png. 4 6 can be something of your choice. Better key setups are shrinking being just alt. Win is for snapping normally. and then just CTRL for the new nine combinations. Shift for scooting. You can also substitute both numpad5's. CTRL/shift/div + 5 will put window into priority, and Win + 5 will maximize and unmaximize. This gives a lot of power to all the Win keys, with CTRL being less used but still very good, such as priority and thirds.

**Comment by glenviewjeff** [comment]
Sorry I never got back to you on this; I think I was pretty confused with all of the detail. Do you still want help with this?

**Comment by geoxcaliber** [comment]
Yes. Let me know if you need any other information.
